### PR TITLE
feature / #597 Make Header sticky

### LINKED ISSUE #597: [FEATURE] Make SfHeader sticky
via `sticky` prop

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue
Closes #597 

# Scope of work
Added sticky prop for SfHeader

# Screenshots of visual changes

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Comment by filrak** [comment]
Fixed is not enough ;) Sorry for lack of info
It needs to 
- become sticky only on scroll
- show only after scrol-up, so if we are scrolling down from top it's not visible, after scrolling up it appears

This is currently most common behavior for stuff like that

**Comment by pspaczek** [comment]
@aniamusial we should do that like on vue-storefront https://demo.storefrontcloud.io/

**Comment by pspaczek** [comment]
SfHeader stories is outdated :(

File: packages/vue/src/components/organisms/SfHeader/SfHeader.js [hunk 1]
```diff
@@ -60,5 +64,48 @@ export default {
...
+  data() {
+    return {
+      isVisible: true,
+      isScrolling: false,
+      scrollTop: 0,
+      lastScrollTop: 0
+    };
+  },
+  beforeMount() {
+    window.addEventListener(
```
Comment (pspaczek) [inline packages/vue/src/components/organisms/SfHeader/SfHeader.js]: If you register event please unregister it on `beforeDestory`
If `sticky=false` you shouldn't fire sticky functions :)

**Comment by pspaczek** [comment]
thanks @aniamusial I will finish this task in #739

**Pull request closed without merging**